### ISSUE TITLE
'utf-8' codec can't decode byte 0xa0 in position 16

### ISSUE THREAD

**Issue by innovate-invent** [galaxyproject/galaxy#12170]
Uploading the attached file to Galaxy 21.01 results in an error while sniffing the file.
It is concerning that the sniffer is even running given that the type was explicitly set as "fasta" before starting the upload.
The upload was done with the web uploader.

```
Traceback (most recent call last):
  File "/srv/galaxy/tools/data_source/upload.py", line 339, in <module>
    __main__()
  File "/srv/galaxy/tools/data_source/upload.py", line 332, in __main__
    metadata.append(add_file(dataset, registry, output_path))
  File "/srv/galaxy/tools/data_source/upload.py", line 131, in add_file
    stdout, ext, datatype, is_binary, converted_path = handle_upload(
  File "/srv/galaxy/lib/galaxy/datatypes/upload_util.py", line 70, in handle_upload
    if check_content and hasattr(datatype, 'sniff') and not datatype.sniff(path):
  File "/srv/galaxy/lib/galaxy/datatypes/sniff.py", line 636, in auto_sniff
    return self.sniff_prefix(file_prefix)
  File "/srv/galaxy/lib/galaxy/datatypes/sequence.py", line 350, in sniff_prefix
    fh = file_prefix.string_io()
  File "/srv/galaxy/lib/galaxy/datatypes/sniff.py", line 580, in string_io
    raise self.non_utf8_error
  File "/srv/galaxy/lib/galaxy/datatypes/sniff.py", line 557, in __init__
    contents_header = contents_header_bytes.decode("utf-8")
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xa0 in position 16: invalid start byte
```

[PAI-CFT073-asnt_(HPI).fasta.txt](https://github.com/galaxyproject/galaxy/files/6679785/PAI-CFT073-asnt_.HPI.fasta.txt)

Job messages:
```
{ "code_desc": "", "desc": "Fatal error: Exit code 1 ()", "error_level": 3, "exit_code": 1, "type": "exit_code" }
```

**Comment by mvdbeek** [comment]
That's a duplicate of https://github.com/galaxyproject/galaxy/issues/5250
